activerecord/lib/active_record/reflection.rb

```diff
@@ -609,12 +608,14 @@ def calculate_constructable(macro, options)
         # If it cannot find a suitable inverse association name, it returns
         # +nil+.
         def inverse_name
-          options.fetch(:inverse_of) do
-            @automatic_inverse_of ||= automatic_inverse_of
+          unless defined?(@automatic_inverse_of)
+            @automatic_inverse_of = options.fetch(:inverse_of) { automatic_inverse_of }
```

Comment:
Prefer ivar name `@inverse_name` because the variable is not only assigned the result of `automatic_inverse_of` and the `inverse_name` method always returns the variable.